src/common/reactNativeProjectHelper.ts

```diff
@@ -10,53 +10,112 @@ import {Package} from "./node/package";
-        return new Package(projectRoot).dependencyPackage("react-native").version()
-            .then(version => ReactNativeProjectHelper.processVersion(version))
-            .catch(err => {
-                throw ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
-            });
+    public static getReactNativePackageVersionsFromNodeModules(projectRoot: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        let versionPromises: Q.Promise<string>[] = [];
+
+        versionPromises.push(
+            new Package(projectRoot).getPackageVersionFromNodeModules("react-native")
+                .then(version => ReactNativeProjectHelper.processVersion(version, true))
+                .catch(err => {
+                    throw ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
+                })
+            );
+
+        if (isRNWindows) {
+            versionPromises.push(
+                new Package(projectRoot).getPackageVersionFromNodeModules("react-native-windows")
+                    .then(version => ReactNativeProjectHelper.processVersion(version, false))
+                    .catch(err => "")
+            );
+        }
+
+        return Q.all(versionPromises).then(packageVersions => ({
+            reactNativeVersion: packageVersions[0] || "",
+            reactNativeWindowsVersion: packageVersions[1] || "",
+        }));
     }
 
-    public static getReactNativeVersionFromProjectPackage(cwd: string): Q.Promise<string> {
+    public static getReactNativeVersionsFromProjectPackage(cwd: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        let parsedPackageNames: ParsedPackageName[] = [
+            {
+                packageName: "react-native",
+                useSemverCoerce: true,
+            },
+        ];
+
+        if (isRNWindows) {
+            parsedPackageNames.push({
+                packageName: "react-native-windows",
+                useSemverCoerce: false,
+            });
+        }
+
         const rootProjectPackageJson = new Package(cwd);
+
         return rootProjectPackageJson.dependencies()
             .then(dependencies => {
-                if (dependencies["react-native"]) {
-                    return ReactNativeProjectHelper.processVersion(dependencies["react-native"]);
-                }
                 return rootProjectPackageJson.devDependencies()
                     .then(devDependencies => {
-                        if (devDependencies["react-native"]) {
-                            return ReactNativeProjectHelper.processVersion(devDependencies["react-native"]);
-                        }
-                        return "";
+                        let versionPromises: Q.Promise<string>[] = [];
+                        parsedPackageNames.forEach(parsedPackageName => {
+                            versionPromises.push(
+                                Q.Promise<string>((resolve, reject) => {
```

Comment:
I have a couple of questions about this code:
1. Do we need to handle already collected versions as an async promises?
1. Can we parse devdependencies only in cases if not all needed versions was found in dependencies? From my understanding it should work that way: 
* If we found needed version in dependencies - we save it
* If not found, trying to find in devdependencies, if found - save it
* If not found - return empty string
So by this algorithm we don't need to parse devdependencies every time since we can find what we need on the first stage.